### PR TITLE
fix pretraining initialization

### PR DESCRIPTION
**Pull request by svlandeg**
## Description
When running `pretrain`, the `nlp` object should be initialized with the relevant train corpus to ensure the initialization of components happens correctly. Otherwise, the tagger for instance may complain about not having any labels.

### Types of change
bug fix

## Checklist
- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by ines** [comment]
This is tricky, though, because it'd mean that the pretraining will _also_ need to have some arbitrary training data available that's not actually used (in addition to the `pretrain` corpus that _is_ used for pretraining). So unless I'm missing something, this is potentially very confusing 🤔 Should we even need to create any components that are not the pretrained component specified in the config?

**Comment by svlandeg** [comment]
Maybe there's a better way around it. But right now we're fetching the `tok2vec` sublayer of the component we want to pretrain, and to be able to get that we need to initialize that component, right?

**Comment by svlandeg** [comment]
Rewrote entirely and reopened the PR as https://github.com/explosion/spaCy/pull/7342
